src/code/z_demo.c

```diff
@@ -1,99 +1,1649 @@
+
+            case CS_CMD_TERMINATOR:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    Cutscene_Command_Terminator(globalCtx, csCtx, (CsCmdBase*)cutscenePtr);
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case CS_CMD_15F:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    func_800EB6F8(globalCtx, csCtx, (CsCmdBase*)cutscenePtr);
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case CS_CMD_TEXTBOX:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    cmd = (CsCmdBase*)cutscenePtr;
+                    if (cmd->base != 0xFFFF) {
+                        Cutscene_Command_Textbox(globalCtx, csCtx, (CsCmdTextbox*)cutscenePtr);
+                    }
+                    cutscenePtr += sizeof(CsCmdTextbox);
+                }
+                break;
+
+            case CS_CMD_SCENE_TRANS_FX:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    Cutscene_Command_TransitionFX(globalCtx, csCtx, (CsCmdBase*)cutscenePtr);
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case CS_CMD_MOTIONBLUR:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    Cutscene_Command_MotionBlur(globalCtx, csCtx, (CsCmdBase*)cutscenePtr);
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case CS_CMD_GIVETATL:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    Cutscene_Command_GiveTatlToPlayer(globalCtx, csCtx, (CsCmdBase*)cutscenePtr);
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case -2:
+                break;
+
+            default:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+        }
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/Cutscene_ProcessCommands.s")
+#endif
+
+void func_800ED980(GlobalContext* globalCtx, CutsceneContext* csCtx) {
+    if (gSaveContext.cutscene >= 0xFFF0) {
+        csCtx->frames++;
+        Cutscene_ProcessCommands(globalCtx, csCtx, (void*)globalCtx->csCtx.csData);
```

Comment:
```suggestion
        Cutscene_ProcessCommands(globalCtx, csCtx, globalCtx->csCtx.csData);
```
I don't think its necessary. Up to you though.